dev/src/components/teacher/EditCriterion.tsx

```diff
@@ -1,36 +1,40 @@
-import { json } from "overmind"
 import React, { useState } from "react"
 import { Assignment, GradingCriterion } from "../../../proto/ag/ag_pb"
+import { Converter } from "../../convert"
 import { useActions } from "../../overmind"
 
 
-const EditCriterion = ({ criterion, benchmarkID, assignment }: { criterion?: GradingCriterion, benchmarkID: number, assignment: Assignment }): JSX.Element => {
+const EditCriterion = ({ criterion, benchmarkID, assignment }: { criterion?: GradingCriterion.AsObject, benchmarkID: number, assignment: Assignment.AsObject }): JSX.Element => {
     const actions = useActions()
 
     const [editing, setEditing] = useState<boolean>(false)
     const [add, setAdd] = useState<boolean>(criterion ? false : true)
-    const [newCriterion, setNewCriterion] = useState<GradingCriterion>(new GradingCriterion().setBenchmarkid(benchmarkID))
-    // Clone the criterion to use as backup in case of cancel
-    const copy = json(criterion)?.cloneMessage()
-    const c = json(criterion) ?? newCriterion
+
+    // Clone the criterion, or create a new one if none was passed in
+    const c = criterion
```

Comment:
Could this be named `newCriterion` instead of `c`? (Since `c.description` is not obvious when used later.)

An alternative could be to rename the input argument to `originalCriterion` and `c` to `criterion`.